matchzoo/preprocessor/dssm_preprocessor.py

```diff
@@ -145,26 +146,31 @@ def transform(
         if stage == 'train':
             # use cached data to fit word hashing layer directly.
             for idx, tri_letter in tqdm(self._cache):
-                outputs.append((idx, hashing.transform(tri_letter)))
-
-            return self._make_output(output=outputs,
-                                     mapping=self._datapack.mapping,
-                                     context=self._context,
-                                     stage=stage)
+                outputs[idx] = hashing.transform(tri_letter)
+
+            return self._make_output(
+                output=self._datapack.dataframe,
+                mapping=outputs,
+                context=self._context,
+                stage=stage
+            )
         else:
             # do preprocessing from scrach.
             units = self._prepare_stateless_units()
             units.append(hashing)
             self._datapack = self.segmentation(inputs, stage='test')
 
-            df = self._datapack.dataframe
+            mapping = self._datapack.mapping
 
-            for idx, text in tqdm(zip(df['id'], df['text'])):
+            for idx, text in tqdm(mapping.items()):
+                text = text['text']
                 for unit in units:
                     text = unit.transform(text)
-                outputs.append((idx, text))
-
-            return self._make_output(output=outputs,
-                                     mapping=self._datapack.mapping,
-                                     context=self._context,
-                                     stage=stage)
+                outputs[idx] = text
+
+            return self._make_output(
+                output=self._datapack.dataframe,
```

Comment:
same, it will pass a `dataframe` as `data` field into datapack.

And inside `datapack` constructor, it will call:
```python
self._dataframe = pd.DataFrame(self._datapack.dataframe)
```